Makefile

```diff
@@ -19,16 +17,20 @@ eventloop.a: $(OBJFILES)
 	ar rcs eventloop.a $(OBJFILES)
 	
 eventloop.o: $(LSOURCES) 
-	$(CXX) -c $(CXXFLAGS)  elooplib/eventloop.cpp
+	$(CXX) -c $(CXXFLAGS) elooplib/eventloop.cpp
 	
-client.o: elooplib/client.cpp elooplib/node.cpp elooplib/threadpool.hpp elooplib/archive.hpp
+client.o: elooplib/client.cpp elooplib/node.cpp elooplib/threadpool.cpp elooplib/archive.hpp
 	$(CXX) -c $(CXXFLAGS) elooplib/client.cpp
 	
 server.o: elooplib/server.cpp elooplib/node.cpp
```

Comment:
No need to recompile server.cpp if node.cpp changes. These files are independent. Make it depend on `node.hpp` instead.

Remember, first you compile every cpp file to o file, then link all o files to single executable. Header files don't get compiled, they are only included to cpp files and usually have only declaration.